Sources/Confidence/Confidence.swift

```diff
@@ -8,26 +8,62 @@ public class Confidence: ConfidenceEventSender {
     public var region: ConfidenceRegion
     public var initializationStrategy: InitializationStrategy
     private var removedContextKeys: Set<String> = Set()
+    private var client: ConfidenceClient
 
-    required public init(
+    required init(
         clientSecret: String,
         timeout: TimeInterval,
         region: ConfidenceRegion,
         initializationStrategy: InitializationStrategy,
         context: ConfidenceStruct = [:],
+        client: ConfidenceClient,
         parent: ConfidenceEventSender? = nil
     ) {
         self.clientSecret = clientSecret
         self.timeout = timeout
         self.region = region
         self.initializationStrategy = initializationStrategy
         self.context = context
+        self.client = client
         self.parent = parent
     }
 
-    // TODO: Implement actual event uploading to the backend
-    public func send(definition: String, payload: ConfidenceStruct) {
-        print("Sending: \"\(definition)\".\nMessage: \(payload)\nContext: \(context)")
+    public convenience init(
+        clientSecret: String,
+        timeout: TimeInterval,
+        region: ConfidenceRegion,
+        initializationStrategy: InitializationStrategy,
+        context: ConfidenceStruct = [:],
```

Comment:
shouldn't we pass here a `ConfidenceContextProvider` instead? if we pass this context to the child confidence using `withContext`, and if the parent context is changed meanwhile, how can we propagate the parent context in the child?

Reply:
`withContext` is passing `self` as `parent`, so that the child instance can call `getContext` on the parent at any time

Reply:
didn't we want to pass confidenceContextProvider instead of the whole interface of EventSender to the child?